src/api/v1/users/register.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Registration endpoints for users
+ */
+import * as Hapi from 'hapi';
+import * as Boom from 'boom';
+import {
+  id,
+  visitorName,
+  birthYear,
+  email,
+  phoneNumber,
+  postCode,
+  emailConsent,
+  smsConsent,
+  gender,
+  response
+} from './schema';
+import { Visitors, CommunityBusinesses, CbAdmins } from '../../../models';
+import * as QRCode from '../../../services/qrcode';
+import * as PdfService from '../../../services/pdf';
+import { EmailTemplate } from '../../../services/email/templates';
+import { RoleEnum } from '../../../auth/types';
+import Roles from '../../../auth/roles';
+
+
+interface RegisterRequest extends Hapi.Request {
+  payload: {
+    organisationId: number
+    name: string
+    gender: string
+    birthYear: number
+    email: string
+    phoneNumber: string
+    postCode: string
+    emailConsent: boolean
+    smsConsent: boolean
+  };
+}
+
+
+export default [
+  {
+    method: 'POST',
+    path: '/users/register/visitor',
+    options: {
+      auth: false,
+      validate: {
+        payload: {
+          organisationId: id.required(),
+          name: visitorName.required(),
+          gender: gender.required(),
+          birthYear: birthYear.required(),
+          email: email.required(),
+          phoneNumber: phoneNumber.allow(''),
+          postCode: postCode.allow(''),
+          emailConsent: emailConsent.default(false),
+          smsConsent: smsConsent.default(false),
+        },
+      },
+      response: { schema: response },
+    },
+    handler: async (request: RegisterRequest, h: Hapi.ResponseToolkit) => {
+      const { knex, payload, server: { app: { EmailService } } } = request;
+
+      /*
+       * Preliminaries
+       * (Can possibly eventually be removed into pre-requisites)
+       */
+      // Check user doesn't already exist
+      if (await Visitors.exists(knex, { where: { email: payload.email } })) {
+        throw Boom.conflict('User with this e-mail already registered');
+      }
+      if (!(await CommunityBusinesses.exists(knex, { where: { id: payload.organisationId } }))) {
+        throw Boom.badRequest('Unrecognised organisation');
+      }
+
+      // Fetch the community business admin
+      const cb = await CommunityBusinesses.getOne(knex, {
+        where: { id: payload.organisationId, deletedAt: null },
+      });
+      const [admin] = await CbAdmins.fromOrganisation(knex, { id: payload.organisationId });
```

Comment:
What's the benefit of this check?

Reply:
I need the email of the associated admin